src/lib/control/v1/kms.js

```diff
@@ -23,15 +27,30 @@ module.exports = () => (req, res, next) => {
     region
   };
 
-  return KMS.encrypt(params).then((data) => {
-    res.send(
-      '$tokend:\n' +
-      '  type: kms\n' +
-      '  resource: /v1/kms/decrypt\n' +
-      '  region: ' + data.region + '\n' +
-      '  ciphertext: "' + data.ciphertext + '"\n' +
-      '  datakey: "' + data.datakey + '"\n');
-  }).catch((err) => {
-    next(err);
-  });
+  let data;
+
+  try {
+    data = await encrypt(params);
+  } catch (err) {
+    return next(err);
+  }
+
+  const resp = {
+    region: data.region
```

Comment:
If `data` is an `Error` (which we're checking for below) will it have a `region` property?

Reply:
Yes. `encrypt` stuffs `region` in the error [here](https://github.com/rapid7/acs/blob/refactor/src/lib/kms/index.js#L51).

The reason I'm doing it this way is that when the work to update the config structure to support multiple key selection is added we need a way to use `Promise.all` to coordinate the calls without failing fast which is its behavior. So, instead of `throw`ing, `encrypt` returns the error object nicely which means that when we handle the results of `Promise.all` we can do a type check and then send the response accordingly.

Reply:
I could add some validation for the POST body to make sure that the key object will always have a `region` and `key`.